src/client.rs

```diff
@@ -345,7 +348,26 @@ impl Client {
             {
                 return Ok(());
             }
-            tokio::time::sleep(Duration::from_secs(5)).await;
+
+            // When backoff returns `None` the timeout has expired
+            match backoff_strategy.next_backoff() {
+                Some(backoff) => {
+                    info!(
+                        "Waiting [{}] seconds before trying again..",
+                        backoff.as_secs()
+                    );
+                    tokio::time::sleep(backoff).await;
+                }
+                None => {
+                    // We offer no way of specifying a timeout, so this shouldn't happen,
+                    // if it does we'll log an error for now and continue iterating and wait for
+                    // the last interval we saw
+                    error!(
+                        "Waiting for deletion timed out, but no timeout was specified, this is an error and should not happen!"
```

Comment:
Looking at ExponentialBackoff::default it defaults to "expire" after 15min: https://github.com/ihrwein/backoff/blob/master/src/default.rs#L13

I'm not sure if you want to change the docs or the behavior or whether having this backoff crate is overkill.
I'm fine with either of those solutions (or another one entirely)

Reply:
Ups, sorry, I missed that. I'll change the default behavior no never expire.

I considered whether we should add that crate, yes. But I pretty much reimplemented the exact functionality when I did it myself, just with less functionality. And I already found stuff that I wanted to add which would have duplicated even more of that crate.

I looked at the crate and it seemed more or less active, docs looked good and the author says where he took inspiration, which is a fairly established backoff algo by Google.

And with an eye on https://github.com/stackabletech/operator-rs/issues/147 I figured it might be worth pulling this in.